experimental/ComplexReflectionGroups/src/hermitian_things.jl

```diff
@@ -0,0 +1,81 @@
+# This file implements some basics to work with the Hermitian inner product over
+# number fields.
+
+# Ulrich Thiel, 2024
+
+export is_unitary
+export is_orthogonal
+
+
+###########################################################################################
+# Complex conjugation
+###########################################################################################
+# In Hecke v0.30.2 complex_conjugation was extended to more fields following my suggestion.
+# I will only do one minor extension following Tommy Hofmanns suggestion:
+complex_conjugation(K::QQAbField) = conj
+
+###########################################################################################
+# Hermitian scalar product
+###########################################################################################
+function scalar_product(v::AbstractAlgebra.Generic.FreeModuleElem{T}, w::AbstractAlgebra.Generic.FreeModuleElem{T}) where T <: QQAlgFieldElem
+
+    V = parent(v)
+    K = base_ring(V)
+    n = dim(V)
+    s = zero(K)
+    conj = complex_conjugation(K)
+    for i=1:n
+        s += v[i]*conj(w[i])
+    end
+    return s
```

Comment:
```suggestion
    @req parent(v) === parent(w) "Incompatible vector spaces"
    conj = complex_conjugation(K)
    return sum(vi * conj(wi) for (vi, wi) in zip(v, w); init=zero(base_ring(v)))
```
parent check missing.
furthermore some suggestion for cleanup

Reply:
Ok for the parent check. Otherwise, I find this much more complicated to read. Is this supposed to be faster?

Reply:
It may be, dependent on the underlying implementation. I particular, in your iterated addition, for every addition a new variable containing the result needs to be allocated

Reply:
Hm, interesting. Is this written/explained somewhere? Given that `scalar_product` may indeed be evaluated many times, I'll do your suggested change.

Reply:
Your code does not work:

```
julia> V = vector_space(QQ,2)
v =Vector space of dimension 2 over rational field

julia> v = V([1,1])
(1, 1)

julia> K = base_ring(v)
Rational field

julia> conj = complex_conjugation(K)
Identity map
  of rational field

julia> sum(vi * conj(wi) for (vi, wi) in zip(v, v); init=zero(K))
ERROR: MethodError: no method matching iterate(::AbstractAlgebra.Generic.FreeModuleElem{QQFieldElem})
```

Reply:
@lgoettgens ?

Reply:
ok, so apparently vector space elements only support getindex but not iteration. In this case, a workaround here would be `sum(v[i] * conj(w[i]) for i in 1:dim(parent(v)); init=zero(base_ring(v)))`. This is now basically your initial version but using `sum` instead of iterated addition.

Reply:
Good, thanks.